downloader/src/main/kotlin/vcs/Subversion.kt

```diff
@@ -245,6 +251,37 @@ private class OrtSVNAuthenticationManager : DefaultSVNAuthenticationManager(
     /* passphrase = */ charArrayOf()
 ) {
     private val ortProxySelector = installAuthenticatorAndProxySelector()
+    private val ortAuthenticator = OrtAuthenticator.getDefaultAuthenticator()
+
+    init {
+        authenticationProvider = object : ISVNAuthenticationProvider {
+            override fun requestClientAuthentication(
+                kind: String,
+                svnurl: SVNURL,
+                realm: String,
+                errorMessage: SVNErrorMessage?,
+                previousAuth: SVNAuthentication?,
+                authMayBeStored: Boolean
+            ): SVNAuthentication? {
+                val url = URL(svnurl.toDecodedString())
+
+                val auth = ortAuthenticator?.requestPasswordAuthenticationInstance(
+                    svnurl.host, null, svnurl.port, svnurl.protocol, null, null, url, Authenticator.RequestorType.SERVER
+                ) ?: return null
+
+                return SVNPasswordAuthentication.newInstance(
```

Comment:
Could this be made a single expression by using
 ```
?.let {  auth ->
  SVNPasswordAuthentication.newInstance( 
    auth.userName, auth.password, ...
  )
}
``` 
This would be more functional style, but maybe harder to grasp.

Reply:
Yes, and it also increases nesting / indentation a bit, which is usually something I usually try to avoid for clarity.